engine/src/main/java/com/agiletec/aps/system/common/entity/loader/AttributeRolesLoader.java

```diff
@@ -61,7 +70,32 @@ private void loadDefaultRoles(String attributeRolesFileName, Map<String, Attribu
 			_logger.error("Error loading attribute Roles : file {}", attributeRolesFileName, t);
 		}
 	}
-	
+    
+    public void loadLocalRoles(String configItemName, Map<String, AttributeRole> defaultCollection) {
+        try {
+            ConfigInterface configManager = this.getBeanFactory().getBean(ConfigInterface.class);
+			String xml = configManager.getConfigItem(configItemName);
+            if (StringUtils.isBlank(xml)) {
+                return;
+            }
+            AttributeRoleDOM dom = new AttributeRoleDOM();
+			Map<String, AttributeRole> attributeRoles = dom.extractRoles(xml, "configItem:" + configItemName);
+			List<AttributeRole> roles = new ArrayList<>(attributeRoles.values());
+			for (int i = 0; i < roles.size(); i++) {
+				AttributeRole role = roles.get(i);
+				if (defaultCollection.containsKey(role.getName())) {
+					_logger.warn("You can't override existing attribute role : {} - {}", role.getName(), role.getDescription());
+				} else {
+					defaultCollection.put(role.getName(), role);
+					_logger.info("Added new attribute role : {} - {}",role.getName(), role.getDescription());
+				}
+                role.setLocal(true);
+			}
+		} catch (EntException | BeansException e) {
+			_logger.error("Error loading local attribute Roles", e);

```

Comment:
can be useful to log also some parameters? `configItemName`?

Reply:
Done